File: Olympiad/COCI/official/2008/contest1_solutions/peg.cpp
#include <cstdio>

char a[7][7+1];

int main( void ) {
   for( int r = 0; r < 7; ++r ) gets( a[r] );

   int ret = 0;
   for( int r = 0; r < 7; ++r ) 
      for( int c = 0; c < 7; ++c ) {
         if( r+2 <  7 && a[r][c] == 'o' && a[r+1][c] == 'o' && a[r+2][c] == '.' ) ++ret;
         if( r-2 >= 0 && a[r][c] == 'o' && a[r-1][c] == 'o' && a[r-2][c] == '.' ) ++ret;
         if( c+2 <  7 && a[r][c] == 'o' && a[r][c+1] == 'o' && a[r][c+2] == '.' ) ++ret;
         if( c-2 >= 0 && a[r][c] == 'o' && a[r][c-1] == 'o' && a[r][c-2] == '.' ) ++ret;
      } 
        
   printf( "%d\n", ret );

   return 0;
}

File: Olympiad/COCI/official/2008/contest1_solutions/peg.pas
program peg;

var
   a : array[1..7] of string;
   r, c : longint;
   ret : longint;

begin
   for r := 1 to 7 do readln( a[r] );

   ret := 0;
   for r := 1 to 7 do
      for c := 1 to 7 do begin
         if (r+2 <= 7) and (a[r][c] = 'o') and (a[r+1][c] = 'o') and (a[r+2][c] = '.') then ret := ret+1;
         if (r-2 >= 1) and (a[r][c] = 'o') and (a[r-1][c] = 'o') and (a[r-2][c] = '.') then ret := ret+1;
         if (c+2 <= 7) and (a[r][c] = 'o') and (a[r][c+1] = 'o') and (a[r][c+2] = '.') then ret := ret+1;
         if (c-2 >= 1) and (a[r][c] = 'o') and (a[r][c-1] = 'o') and (a[r][c-2] = '.') then ret := ret+1;
      end;

   writeln( ret );
end.
